mongo_connector/doc_managers/algolia_doc_manager.py

```diff
@@ -250,7 +250,8 @@ def apply_update(self, doc, update_spec):
                     doc[attr] = None
         return doc
 
-    def update(self, doc, update_spec, namespace = None, timestamp = None):
+    def update(self, document_id, update_spec, namespace = None, timestamp = None):
+        doc = self.index.getObject(str(document_id))
```

Comment:
absolutely. well maybe if the PUT to algolia index supported accepts a partial document that could be avoided, but yes, I'm not sure the DocManagerBase was designed to support that anyway.